### PR TITLE
Test shippingOption set correctly

### PR DESCRIPTION
**Pull request by marcoscaceres**
The spec does a weird thing that if `requestShipping` is false, but shipping options are given, the request returns the selected shipping option, but the response.shippingOption will be null.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-09-04 06:44:57
Finished: 2017-09-04 06:51:18


<h2>Failing Jobs</h2>
<ul>

<li>chrome:unstable</li>

<li>MicrosoftEdge:14.14393</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/18136)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/271570509)

File: payment-request/payment-response/helpers.js [hunk 1]
```diff
@@ -75,5 +75,24 @@ async function runManualTest(button, options, expected = {}, id = undefined) {
         `Expected response ${attribute} attribute to be ${value}`
       );
     }
+    // Special handling for option.requestShipping
+    assert_equals(
```
Comment (edenchuang) [inline payment-request/payment-response/helpers.js]: Maybe a spec issue here, for the case that giving shippingOptions when options.requestShipping is false.
Do we have CR for this?

**Review by edenchuang — APPROVED** [review]
We probably have spec issue here. Please send a CR to trace this issue.

File: payment-request/payment-response/helpers.js [hunk 2]
```diff
@@ -108,5 +108,23 @@ async function runManualTest(button, options, expected = {}, id = undefined) {
...
+    } else {
+      assert_equals(
+        request.shippingOption,
+        null,
+        "If requestShipping is falsy, request.shippingOption must be null"
+      );
+      assert_equals(
+        response.shippingOption,
+        null,
+        "request.shippingOption must be 'pass'"
```
Comment (edenchuang) [inline payment-request/payment-response/helpers.js]: s/pass/null

File: payment-request/payment-request-constructor.https.html [hunk 3]
```diff
@@ -368,21 +376,46 @@
   assert_equals(
     request.shippingOption,
     null,
-    "request.shippingOption must be null"
+    "shippingOption must be null, as requestShipping is missing"
```
Comment (edenchuang) [inline payment-request/payment-request-constructor.https.html]: This is inconsistent with test behavior. Might append the original test below the assertion check.
`
const request2 = new PaymentRequest(defaultMethods, details, {requestShipping: true});
assert_equals( request.shippingOption, null, `request.shippingOption must be null`);
`

**Review by edenchuang — COMMENTED** [review]
Some nits to fix

**Comment by marcoscaceres** [comment]
Sorry these slipped through. Sent a new PR: https://github.com/w3c/web-platform-tests/pull/7289

**Comment by AmazingJaze** [comment]
//CC @marcoscaceres is there an ETA on when the spec will be updated to match the behavior in this test?

**Comment by marcoscaceres** [comment]
@AmazingJaze done! Make sure you do a "shift-refresh", as w3c has fairly aggressive caching. You should see version: "W3C Working Draft 11 September 2017"

**Comment by marcoscaceres** [comment]
Or, "W3C Editor's Draft 11 September 2017"... if you look at the Editor's draft: https://w3c.github.io/payment-request/

The Editor's draft is always more authoritative.

**Pull request merged**